tests/components/iskra/test_config_flow.py

```diff
@@ -0,0 +1,241 @@
+        self.serial = SERIAL
+        self.model = model
+        self.description = "Iskra mock device"
+        self.location = "imagination"
+        self.sw_ver = "1.0.0"
+
+
+@pytest.fixture
+def mock_pyiskra_rest():
+    """Mock Iskra API authenticate with Rest API protocol."""
+
+    with patch(
+        "pyiskra.adapters.RestAPI.RestAPI.get_basic_info",
+        return_value=MockBasicInfo(model=SG_MODEL),
+    ) as basic_info_mock:
+        yield basic_info_mock
+
+
+@pytest.fixture
+def mock_pyiskra_modbus():
+    """Mock Iskra API authenticate with Rest API protocol."""
+
+    with patch(
+        "pyiskra.adapters.Modbus.Modbus.get_basic_info",
+        return_value=MockBasicInfo(model=PQ_MODEL),
+    ) as basic_info_mock:
+        yield basic_info_mock
+
+
+# Test step_user with Rest API protocol
+async def test_user_rest_no_auth(hass: HomeAssistant, mock_pyiskra_rest) -> None:
+    """Test the user flow with Rest API protocol."""
+
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    # Test no authentication required
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: "rest_api"},
+    )
+
+    # Test successful Rest API configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{SG_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
+
+
+async def test_user_rest_auth(hass: HomeAssistant, mock_pyiskra_rest) -> None:
+    """Test the user flow with Rest API protocol and authentication required."""
+    mock_pyiskra_rest.side_effect = NotAuthorised
+
+    # Test if prompted to enter username and password if not authorised
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: "rest_api"},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "authentication"
+
+    # Test failed authentication
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_USERNAME: "", CONF_PASSWORD: ""},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+    assert result["step_id"] == "authentication"
+
+    # Test successful authentication
+    mock_pyiskra_rest.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_USERNAME: "", CONF_PASSWORD: ""},
+    )
+
+    # Test successful Rest API configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{SG_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
```

Comment:
idem

Reply:
added full data dict asserts

Reply:
Also change this to the way I explained above